### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ClementPinard**
- remove numpy and scipy pin
- remove fiftyone-db-22.04 pin

Update code to be compatible with numpy 2.0 :

- Replace np.NaN with np.nan
- Replace numpy scalars with the python version for the label map, for presentation purposes

### CODE REVIEW

File: lours/dataset/dataset.py [hunk 1]
```diff
@@ -1006,7 +1006,12 @@ def init_annotations(self):
                 f" their string equivalent : {missing_ids}"
             )
             for i in missing_ids:
-                self.label_map[i] = str(i)
```
Comment (NacerKaciXXII) [inline lours/dataset/dataset.py:1009]: Do you know why Codecov is saying that this is not covered by the tests? If it was covered before, I don't see how this new change is not

**Review by NacerKaciXXII — COMMENTED** [review]
I did the following checks on Ubuntu 20.04 and Ubuntu 22.04:
- Run pytest => All tests are passing
- Load a dataset to fiftyone and run the fiftyone app => works fine even without the fifityone-db-22.04

If you see other necessary tests let me know and I can do them. If not, then there is only my comment on Codecov's coverage regression and I'm good for this MR!

**Comment by ClementPinard** [comment]
Ok took me some time to understand, but I now understand:
codecov complains that these 2 new lines are not covered by tests. Codecov default rule is that coverage can only go up, so the coverage of newly added lines must be above current coverage (which is 86%)

Why are these two lines never reached ? Because since we are always dealing with numpy arrays, the `.item()` method will always work. Even if I try to put something else like I do in the new test (`my_dummy_dataset.annotations.loc[my_dummy_dataset.annotations.index[0], "category_id"] = 1`) it gets converted back to int64 with this line : https://github.com/XXII-AI/Lours/blob/2f9d16614072c16f9924bd2b53896a8cf70733f3/lours/dataset/dataset.py#L1018

An interesting issue would be to discuss the possibility to have flexible dtypes. For example, you could use category_id with strings, especially uuids.

We'll see about that in the future. In the mean time, I just remove this try except since it cannot be triggered by design

**Comment by ClementPinard** [comment]
Just updated the branch to also have the changelog changed, we need to add an action to check it everytime :p

**Review by NacerKaciXXII — APPROVED** [review]
Thanks for the codecov investigation!

**Pull request merged**